### PR TITLE
[XPU] fix batch_norm_grad when use global status

### PR DESCRIPTION
**Pull request by zhouquan32**
### PR Category
Custom Device

### PR Types
Bug fixes

### Description
There is an error in the XPU API related to batch_norm_grad when using global mean and global variance to modify batch normalization. The original implementation may cause unexpected situations, such as the XPU hanging. To address this issue, consider using a combination of alternative APIs to replace the original process.

### CODE REVIEW

File: paddle/phi/kernels/xpu/batch_norm_grad_kernel.cc [hunk 1]
```diff
@@ -192,34 +192,32 @@ void BatchNormGradKernel(const Context &dev_ctx,
   const auto *global_mean = mean.get_ptr();
   const auto *global_var = variance.get_ptr();
```
Comment (skywalker2012) [inline paddle/phi/kernels/xpu/batch_norm_grad_kernel.cc:193]: 当use_global_stats=True时，建议加个global_mean和global_var非空检查，类似：
```
PADDLE_ENFORCE_NOT_NULL(global_mean, ...);  
PADDLE_ENFORCE_NOT_NULL(global_var, ...);
```

**Comment by dynamicheart** [comment]
需要增加单测

**Review by dynamicheart — APPROVED** [review]
LGTM

**Comment by skywalker2012** [comment]
LGTM

**Review by QingshuChen — APPROVED** [review]
